docs/source/library-user-guide/upgrading.md

```diff
@@ -0,0 +1,65 @@
+<!---
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# Upgrade Guides
+
+## DataFusion `46.0.0`
+
+
+### Changes to `invoke()` and `invoke_batch()` deprecated
+
+We are migrating away from `ScalarUDFImpl::invoke()` and
```

Comment:
I added this section based on the report from @shehabgamin in https://github.com/apache/datafusion/issues/14123#issuecomment-2684612667

@shehabgamin any chance you have an example of the kind of error you saw?

Reply:
Thank you so much @shehabgamin  -- I have incorporated this content in 9c658c739